catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +251,109 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
         return metatypes;
     }
 
+    @Override
+    public List<Serializable> updatePublications(String source, List<String> destinations)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        List<Serializable> updatedPublishedLocations = new ArrayList<>();
+
+        if (destinations != null) {
+
+            Filter filter = filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .is()
+                    .equalTo()
+                    .text(source);
+            Query query = new QueryImpl(filter);
+
+            QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(query));
+
+            List<Result> metacards = queryResponse.getResults();
+            if (metacards != null && metacards.size() > 0) {
+                Metacard metacard = metacards.get(0)
+                        .getMetacard();
+                if (metacard != null) {
+                    List<Serializable> currentlyPublishedLocations = metacard.getAttribute(
+                            RegistryObjectMetacardType.PUBLISHED_LOCATIONS)
+                            .getValues();
+
+                    List<String> unpublishLocations = new ArrayList<>();
+
+                    unpublishLocations = currentlyPublishedLocations.stream()
+                            .filter(location -> !destinations.contains((String) location))
+                            .map(location -> (String) location)
+                            .collect(Collectors.toList());
+
```

Comment:
Think we need to create a list of location to publish to so we don't try to publish to places we already have published. You could use the same logic as above but switch currentlyPublishedLocations and destinations. Then loop over that list below.